src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -325,6 +326,11 @@ void reconnect() throws InterruptedException {
             if (maxTries > 0 && tries >= maxTries) {
                 break;
             }
+
+            if (shouldWait) {

```

Comment:
shouldn't this happen inside the for loop? line 318 when we did not succeed or exceed the limits?

Reply:
it is in the while loop, but i could see it going either way for sure

Reply:
might be cleaner to move it up